### PR TITLE
Dts 28092 Integrate Mongock and update DDL script

### PR DESCRIPTION
**Pull request by hargupta15**
1. Integrated Mongock
2. Updated DDL command

### CODE REVIEW

File: customapi/src/main/resources/json/mongock/default/kpi_master.json [hunk 1]
```diff
@@ -0,0 +1,3668 @@
+[
```
Comment (Chittauri) [inline customapi/src/main/resources/json/mongock/default/kpi_master.json:1]: why cant we use src/test/resources/json/default/kpi_master.json

File: customapi/src/main/java/com/publicissapient/kpidashboard/apis/mongock/upgrade/Release_8_0_0.java [hunk 2]
```diff
@@ -0,0 +1,111 @@
+package com.publicissapient.kpidashboard.apis.mongock.upgrade;
```
Comment (Chittauri) [inline customapi/src/main/java/com/publicissapient/kpidashboard/apis/mongock/upgrade/Release_8_0_0.java:1]: copyright missing

File: customapi/src/main/java/com/publicissapient/kpidashboard/apis/data/FieldMappingStructureDataFactory.java [hunk 3]
```diff
@@ -0,0 +1,82 @@
...
+import com.fasterxml.jackson.databind.SerializationFeature;
+import com.fasterxml.jackson.datatype.jsr310.JavaTimeModule;
+import com.publicissapient.kpidashboard.common.model.application.BaseFieldMappingStructure;
+import lombok.extern.slf4j.Slf4j;
+import org.apache.commons.lang3.StringUtils;
+
+import java.io.IOException;
+import java.util.List;
+
+@Slf4j
```
Comment (Chittauri) [inline customapi/src/main/java/com/publicissapient/kpidashboard/apis/data/FieldMappingStructureDataFactory.java:31]: author, file overview absent

File: customapi/src/main/java/com/publicissapient/kpidashboard/apis/data/KpiCategoryDataFactory.java [hunk 4]
```diff
@@ -0,0 +1,89 @@
...
+import com.fasterxml.jackson.databind.DeserializationFeature;
+import com.fasterxml.jackson.databind.ObjectMapper;
+import com.fasterxml.jackson.databind.SerializationFeature;
+import com.fasterxml.jackson.datatype.jsr310.JavaTimeModule;
+import com.publicissapient.kpidashboard.common.model.application.KpiCategory;
+
+import lombok.extern.slf4j.Slf4j;
+
+/**
+ * @author anisingh4
```
Comment (Chittauri) [inline customapi/src/main/java/com/publicissapient/kpidashboard/apis/data/KpiCategoryDataFactory.java:35]: change author

File: customapi/src/main/java/com/publicissapient/kpidashboard/apis/mongock/ApplicationStartupListener.java [hunk 5]
```diff
@@ -0,0 +1,39 @@
+package com.publicissapient.kpidashboard.apis.mongock;
```
Comment (Chittauri) [inline customapi/src/main/java/com/publicissapient/kpidashboard/apis/mongock/ApplicationStartupListener.java:1]: copyright absent

File: customapi/src/main/java/com/publicissapient/kpidashboard/apis/mongock/upgrade/Release_8_0_0.java (hunk 2 above)
Comment (eswbogol) [reply customapi/src/main/java/com/publicissapient/kpidashboard/apis/mongock/upgrade/Release_8_0_0.java]: added

File: customapi/src/main/java/com/publicissapient/kpidashboard/apis/data/FieldMappingStructureDataFactory.java (hunk 3 above)
Comment (eswbogol) [reply customapi/src/main/java/com/publicissapient/kpidashboard/apis/data/FieldMappingStructureDataFactory.java:33]: added

File: customapi/src/main/java/com/publicissapient/kpidashboard/apis/data/KpiCategoryDataFactory.java (hunk 4 above)
Comment (eswbogol) [reply customapi/src/main/java/com/publicissapient/kpidashboard/apis/data/KpiCategoryDataFactory.java]: changed

File: customapi/src/main/java/com/publicissapient/kpidashboard/apis/mongock/ApplicationStartupListener.java (hunk 5 above)
Comment (eswbogol) [reply customapi/src/main/java/com/publicissapient/kpidashboard/apis/mongock/ApplicationStartupListener.java:17]: added

**Review by eswbogol — APPROVED** [review]

**Review by hirbabar — APPROVED** [review]

**Pull request merged**